docs/usecases/streaming.md

```diff
@@ -3,4 +3,105 @@ id: usecases_streaming
 title:  "Streaming"
 ---
 
-Coming soon...
+All material on this pages assumes knowledge about the `ZIO[R, E, A]` type.
+
+## Why we need streams?
+TBD
+
+## How `zio-streams` are build
+
+The backbone of `zio-streams` are 3 types: `ZStream`, `Pull` and `ZSink`. 
+`ZStream[R, E, A]` represents the stream itself, which produces effectful values of type `A` and might end up with 
+error `E`. The `R` parameter has exactly the same meaning as as in the `ZIO` type that is to represent the environment
+requirements to run this stream.
+Second type, `Pull` should be understood as a handle by which consecutive effectful elements are being pulled into the
+stream (thus the name). `Pull`'s exact definition is a type alias for the `ZIO` type:
+`Pull[-R, +E, +A] = ZIO[R, Option[E], A]`. The contract states that each evaluation of the `Pull` type either produces a
+value `A` which will flow through the stream or produces an `Option[E]` which signals an actual error in `Some(_)` case 
+and flags the end of the stream in the`None` case. To emphasize this - the same `Pull` is being run multiple times to
+get items of the stream produced.
+
+How are `ZStream` and `Pull` related? `ZStream` is constructed based on the `Pull` type - passed as a constructor
+argument wrapped in `ZManaged`. Why the `ZManaged` indirection? Since our `Pull` contract says in most cases we need to assume it
```

Comment:
`ZManaged` is actually needed here to safely allocate and release resources. Would be good to note that.